network/wsNetwork.go

```diff
@@ -823,6 +839,25 @@ func (wn *WebsocketNetwork) Start() {
 	} else {
 		wn.scheme = "http"
 	}
+
+	// if PublicAddress set to automatic, pull the name from Address()
+	if wn.config.PublicAddress == autoconfigPublicAddress {
+		addr, ok := wn.Address()
+		if ok {
+			url, err := url.Parse(addr)
+			if err == nil {
+				wn.config.PublicAddress = fmt.Sprintf("%s:%s", url.Hostname(), url.Port())
```

Comment:
modifying config that has constant/read only semantic in random parts of codebase is not good at all. Why not a local var since all the usage of it is right here?